### PR TITLE
[IMP] website: feedback choose theme

### PR DESCRIPTION
**Pull request by JKE-be**
Move switchView according to FP request

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website/static/src/js/theme_preview_form.js [hunk 1]
```diff
@@ -28,53 +27,49 @@ var ThemePreviewController = FormController.extend({
...
      */
-    autofocus: function () {
-        // force refresh label of button switch
-        this.$switchButton = this._renderSwitchButton();
-        $('.o_switch_mode_button').replaceWith(this.$switchButton);
-        this._super.apply(this, arguments);
+    renderPager: function ($node, options) {
+        return this._super($node, options).then(() => {
+            this.$switcherButton = $(qweb.render('website.ThemePreview.SwitchModeButton'));
+            this.$switcherButton.appendTo(this.pager.$el);
```
Comment (qsm-odoo) [inline addons/website/static/src/js/theme_preview_form.js]: Why not doing this in `_updatePager` ?

File: addons/website/static/src/xml/theme_preview.xml [hunk 2]
```diff
@@ -9,16 +9,19 @@
...
-            <i t-attf-class="fa #{icon or 'fa-refresh'} mr-1"></i>
-            <t t-esc="PreviewType"/> Preview
-        </button>
+        <div class="btn-group btn-group-toggle ml-1" data-toggle="buttons">
+          <label class="btn btn-secondary active">
+            <input type="radio" name="viewer" data-mode='desktop' autocomplete="off" checked='checked'/> Desktop
+          </label>
+          <label class="btn btn-secondary">
+            <input type="radio" name="viewer" data-mode='mobile' autocomplete="off"/> Mobile
+          </label>
```
Comment (qsm-odoo) [inline addons/website/static/src/xml/theme_preview.xml]: I think the indentation is incorrect

File: addons/website/static/src/js/theme_preview_form.js (hunk 1 above)
Comment (JKE-be) [reply addons/website/static/src/js/theme_preview_form.js]: renderPager call the once, whoch add the button
_updatePager called several times that replacing existing button.

But we can do it in _updatePager with one if... if you prefer...

**Comment by JKE-be** [comment]
@qsm-odoo 

Voila c updated

File: addons/website/static/src/js/theme_preview_form.js [hunk 3]
```diff
@@ -28,53 +27,46 @@ var ThemePreviewController = FormController.extend({
...
+        const $buttonSwitch = $(qweb.render('website.ThemePreview.SwitchModeButton'));
+        if (!this.$switcherButton) {
+            this.$switcherButton = $buttonSwitch;
+            this.$switcherButton.appendTo(this.pager.$el);
+        } else {
+            const $switchButton = this.$switcherButton;
+            this.$switcherButton = $buttonSwitch;
+            $switchButton.replaceWith(this.$switcherButton);
+
+        }
```
Comment (qsm-odoo) [inline addons/website/static/src/js/theme_preview_form.js]: You can remove the extra blank line and you can put the duplicated line `this.$switcherButton = $buttonSwitch;` after the if and else. And inside the if and else, use the `$buttonSwitch` variable. The goal is just to have less lines of code (untested):
```javascript
        const $buttonSwitch = $(qweb.render('website.ThemePreview.SwitchModeButton'));
        if (!this.$switcherButton) {
            $buttonSwitch.appendTo(this.pager.$el);
        } else {
            this.$switcherButton.replaceWith($buttonSwitch);
        }
        this.$switcherButton = $buttonSwitch;
```

Sorry if you find that useless, it is just my opinion, you can disagree.

**Comment by JKE-be** [comment]
@qsm-odoo 

can you r+ or send in india for test ;) ?

File: addons/theme_default/data/theme_default_data.xml [hunk 4]
```diff
@@ -1,8 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<odoo>
-
-<record id="website.default_website" model="website">
-    <field name="theme_id" ref="base.module_theme_default"/>
-</record>
-
-</odoo>
```
Comment (qsm-odoo) [inline addons/theme_default/data/theme_default_data.xml]: Should not we remove the file entirely?

Comment (JKE-be) [reply addons/theme_default/data/theme_default_data.xml]: pour moi oui, meme le dossier data et models sont useless...

Comment (qsm-odoo) [reply addons/theme_default/data/theme_default_data.xml]: Y'a plus de dossier model vu que plus de fichier dedans ;) Mais oui, retirons le fichier data donc ;)

**Comment by JKE-be** [comment]
@robodoo r+

**Comment by JKE-be** [comment]
@robodoo rebase-ff

**Pull request closed without merging**